src/System.Devices.Gpio/System.Devices.Gpio.Samples/Program.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System.Diagnostics;
+using System.Threading;
+
+namespace System.Devices.Gpio.Samples
+{
+    internal class Program
+    {
+        private static void Main(string[] args)
+        {
+            try
+            {
+                var option = -1;
+
+                if (args.Length > 0)
+                {
+                    option = Convert.ToInt32(args[0]);
+                }
+
+                switch (option)
+                {
+                    case 0: Unix_BlinkingLED(); break;
+                    case 1: Unix_ButtonLED(); break;
+
+                    case 2: RaspberryPi_BlinkingLED(); break;
+                    case 3: RaspberryPi_ButtonLED(); break;
+
+                    case 4: UnixDriver_BlinkingLED(); break;
+                    case 5: UnixDriver_ButtonLED(); break;
+
+                    case 6: RaspberryPiDriver_BlinkingLED(); break;
+                    case 7: RaspberryPiDriver_ButtonLED(); break;
+
+                    default: RaspberryPiDriver_ButtonLED(); break;
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+            }
+
+            Console.WriteLine("Done!");
+        }
+
+        private static void Unix_BlinkingLED()
+        {
+            Console.WriteLine(nameof(Unix_BlinkingLED));
+            BlinkingLED(new UnixDriver());
+        }
+
+        private static void Unix_ButtonLED()
+        {
+            Console.WriteLine(nameof(Unix_ButtonLED));
+            ButtonLED(new UnixDriver());
+        }
+
+        private static void RaspberryPi_BlinkingLED()
+        {
+            Console.WriteLine(nameof(RaspberryPi_BlinkingLED));
+            BlinkingLED(new RaspberryPiDriver());
+        }
+
+        private static void RaspberryPi_ButtonLED()
+        {
+            Console.WriteLine(nameof(RaspberryPi_ButtonLED));
+            ButtonLED(new RaspberryPiDriver());
+        }
+
+        private static void BlinkingLED(GpioDriver driver)
+        {
+            using (driver)
+            {
+                var led = new GpioPin(driver, GpioScheme.BCM, 26, GpioPinMode.Output);
+
+                for (var i = 0; i < 5; ++i)
+                {
+                    led.Write(GpioPinValue.High);
```

Comment:
Just a thought, but you could probably also add a Toggle API to flip a pin's state.

Reply:
Yes, that's a good idea. I think I will add that kind of useful and handy functionalities as extension methods, since they can be easily implemented using the already existing API.

Reply:
Actually, I don't know if we can read the value of a pin that is configured as output. I will try it on Raspberry Pi, but even if it does maybe other devices work different.